x-pack/plugins/ml/public/application/jobs/jobs_list/components/confirm_modals/edit_job_confirm_modal.tsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState, useEffect, useCallback } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiSpacer,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiText,
+} from '@elastic/eui';
+import { MlSummaryJob } from '../../../../../../common/types/anomaly_detection_jobs';
+import { isManagedJob } from '../../../jobs_utils';
+
+type ShowFunc = (jobs: MlSummaryJob) => void;
+
+interface Props {
+  setShowFunction(showFunc: ShowFunc): void;
+  unsetShowFunction(): void;
+  refreshJobs(): void;
+  showEditJobFlyout(job: MlSummaryJob): void;
+}
+
+export const EditJobConfirmModal: FC<Props> = ({
+  setShowFunction,
+  unsetShowFunction,
+  showEditJobFlyout,
+}) => {
+  const [editing, setEditing] = useState(false);
+  const [modalVisible, setModalVisible] = useState(false);
+  const [isManaged, setIsManaged] = useState(true);
+  const [job, setJob] = useState<MlSummaryJob>();
+
+  useEffect(() => {
+    if (typeof setShowFunction === 'function') {
+      setShowFunction(showModal);
+    }
+    return () => {
+      if (typeof unsetShowFunction === 'function') {
+        unsetShowFunction();
+      }
+    };
+  }, []);
+
+  const showModal = useCallback((item: MlSummaryJob) => {
+    if (isManagedJob(item)) {
+      setModalVisible(true);
+      setEditing(false);
+      setIsManaged(true);
+      setJob(item);
+    }
+  }, []);
+
+  const closeModal = useCallback(() => {
+    setModalVisible(false);
+    setIsManaged(false);
+  }, []);
+
+  if (modalVisible === false) {
+    return null;
+  }
+
+  if (isManaged) {
+    const title = (
+      <FormattedMessage
+        id="xpack.ml.jobsList.editJobModal.editJobsTitle"
+        defaultMessage="Edit {jobId}?"
+        values={{
+          jobId: job?.id,
+        }}
+      />
+    );
+
+    return (
+      <EuiModal data-test-subj="mlEditJobConfirmConfirmModal" onClose={closeModal}>
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>{title}</EuiModalHeaderTitle>
+        </EuiModalHeader>
+        <EuiModalBody>
+          <p>
+            <>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.ml.jobsList.editJobModal.editManagedJobDescription"
+                  defaultMessage="This job has been deployed and managed by Elastic so editing it may impact other parts of the product."
```

Comment:
>                   defaultMessage="This job has been deployed and managed by Elastic so editing it may impact other parts of the product."

This message doesn't seem to match the other ones. Was this intentional?

Reply:
This is because there's no bulk action for editing. The modal will only show for one job.

Reply:
My apologies - I understood you mean now. Updated here  https://github.com/elastic/kibana/pull/122305/commits/427be5b2ff13c2453e9b03611547f978baa19cbc